### PR TITLE
VIR-89 Kustannuspaikka frontend

### PR DESCRIPTION
**Pull request by nniemi**
Added kustannuspaikka autocomplete field for creating new position
Added repository for organizationtree to fetch data from the database

### CODE REVIEW

File: backend/Virkarekisteri/src/Functions/OrganizationTree/GetOrganizationTrees.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+using Virkarekisteri.Repositories;
+
+namespace Virkarekisteri.Functions.OrganizationTree
+{
+    public class GetOrganizationTrees
+    {
+        private readonly ILogger<GetOrganizationTrees> _logger;
+        private readonly IOrganizationTreeRepository _organizationTreeRepository;
+
+        public GetOrganizationTrees(
```
Comment (0x464e) [inline backend/Virkarekisteri/src/Functions/OrganizationTree/GetOrganizationTrees.cs]: Would be nice if we utilized primary constructors here as well to keep the functions in line with all current other functions.

File: backend/Virkarekisteri/src/Functions/OrganizationTree/GetOrganizationTrees.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Logging;
+using Virkarekisteri.Repositories;
+
+namespace Virkarekisteri.Functions.OrganizationTree
+{
```
Comment (0x464e) [inline backend/Virkarekisteri/src/Functions/OrganizationTree/GetOrganizationTrees.cs:13]: This is a nitpick you don't have to fix, but since there is just one namespace in this file, we can use a file-scoped namespace here to avoid unnecessary nesting.

(In plain English, the braces can be removed)

File: backend/Virkarekisteri/src/Models/OrganizationTree.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+namespace Virkarekisteri.Models
+{
+    [Table("OrganizationTree")]
+    public class OrganizationTree
+    {
+        public Guid Id { get; set; }
+
+        [Required]
+        [MaxLength(255)]
+        public string? Name { get; set; }
```
Comment (0x464e) [inline backend/Virkarekisteri/src/Models/OrganizationTree.cs]: Don't make required fields nullable.

File: backend/Virkarekisteri/src/Repositories/OrganizationTreeRepository.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Virkarekisteri.Models;
+
+namespace Virkarekisteri.Repositories
+{
```
Comment (0x464e) [inline backend/Virkarekisteri/src/Repositories/OrganizationTreeRepository.cs:9]: If you switch to a file-scoped namespace as per the earlier comment, it can be done here as well.

File: frontend/src/models/OrganizationTree.ts
Comment (0x464e) [inline frontend/src/models/OrganizationTree.ts:1]: Should required fields also be non-nullable here?

**Review by 0x464e — CHANGES_REQUESTED** [review]

File: backend/Virkarekisteri/src/Functions/OrganizationTree/GetOrganizationTrees.cs (hunk 1 above)
Comment (nniemi) [reply backend/Virkarekisteri/src/Functions/OrganizationTree/GetOrganizationTrees.cs]: Fixed

File: backend/Virkarekisteri/src/Functions/OrganizationTree/GetOrganizationTrees.cs (hunk 2 above)
Comment (nniemi) [reply backend/Virkarekisteri/src/Functions/OrganizationTree/GetOrganizationTrees.cs:13]: Fixed

File: backend/Virkarekisteri/src/Repositories/OrganizationTreeRepository.cs (hunk 4 above)
Comment (nniemi) [reply backend/Virkarekisteri/src/Repositories/OrganizationTreeRepository.cs:9]: Fixed

File: frontend/src/models/OrganizationTree.ts
Comment (nniemi) [reply frontend/src/models/OrganizationTree.ts:1]: Fixed

**Review by 0x464e — APPROVED** [review]
